file-service-persistence/src/main/java/uk/gov/justice/services/fileservice/repository/FileStore.java

```diff
@@ -84,13 +84,14 @@ public UUID store(
             final Optional<FileContent> content = contentJdbcRepository.findByFileId(fileId, connection);
 
             if (!metadata.isPresent()) {
+                close(connection);
```

Comment:
Not really. AFAIK, the connection needs to stay open until the returned stream is consumed. Closing the stream, closes the connection.